frontend_lib/src/component/LinkPreview/LinkPreview.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import { getUrlPreview } from '../../action.async.js'
+import { handleFetchResult } from '../../helper.js'
+
+const MAX_DESCRIPTION_LENGTH = 0 // 0 means no limit
+
+function getDomain (url) {
+  // RJ - NOTE - 2021-03-30
+  // - getDomain("http://perdu.com") if given "perdu.com"
+  // - getDomain("https://example.org/index.html") = "example.org"
+  // - getDomain("https://example.org:443/index.html") = "example.org"
+  // - getDomain("https://example.org:443") = "example.org"
+  // Returns the original URL if the regular expression does not match it.
+  // This function might have its place in helper.js.
+
+  const match = url.match(/^[a-z]+:\/\/([^/:]+)(?::[0-9]+)?(?:\/(?:[\s\S]+)?)?$/i)
+  return (match && match[1]) || url
+}
+
+export default class LinkPreview extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      title: '',
+      url: '',
+      description: ''
+    }
+  }
+
+  getSelectedLink () {
+    const links = (
+      new DOMParser().parseFromString(
+        this.props.findLinkInHTML,
+        'text/html'
+      ).body.querySelectorAll('a[href]')
+    )
+
+    for (const link of links) {
+      const href = link.href
+      if (href.startsWith('https://') && !href.startsWith('https://' + location.hostname)) {
```

Comment:
is it ok if we don't handle http:// protocol?

Reply:
It can be necessary to use http without https. If you use Tracim in internal network (like available in many big company).

Reply:
@raphj

Reply:
This is a functionnal decision, not technical one.

There are still HTTP only websites, so I suggest to allow by default.

An improvment should be to warn the user, an other one to allow/disallow http link preview

Reply:
We can go on like this (without http preview) but we need this kind of decision not being decided by developers but by functionnal team